### PR TITLE
Auth and issue testing

### PR DESCRIPTION
**Pull request by dhrumilp15**

### CODE REVIEW

**Comment by aadibajpai** [comment]
> Hello @dhrumilp15! Thanks for updating this PR.
> 
> * In the file [`test_auth.py`](https://github.com/SwagLyrics/swaglyrics-backend/blob/9ac40293d2ad0b9bebddd95f536a832e8de6f868/test_auth.py):
> 
> > [Line 1:1](https://github.com/SwagLyrics/swaglyrics-backend/blob/9ac40293d2ad0b9bebddd95f536a832e8de6f868/test_auth.py#L1): [F401](https://duckduckgo.com/?q=pep8%20F401) 'unittest' imported but unused
> > [Line 3:1](https://github.com/SwagLyrics/swaglyrics-backend/blob/9ac40293d2ad0b9bebddd95f536a832e8de6f868/test_auth.py#L3): [F403](https://duckduckgo.com/?q=pep8%20F403) 'from auth import *' used; unable to detect undefined names
> > [Line 7:24](https://github.com/SwagLyrics/swaglyrics-backend/blob/9ac40293d2ad0b9bebddd95f536a832e8de6f868/test_auth.py#L7): [E251](https://duckduckgo.com/?q=pep8%20E251) unexpected spaces around keyword / parameter equals
> > [Line 7:26](https://github.com/SwagLyrics/swaglyrics-backend/blob/9ac40293d2ad0b9bebddd95f536a832e8de6f868/test_auth.py#L7): [E251](https://duckduckgo.com/?q=pep8%20E251) unexpected spaces around keyword / parameter equals
> > [Line 9:1](https://github.com/SwagLyrics/swaglyrics-backend/blob/9ac40293d2ad0b9bebddd95f536a832e8de6f868/test_auth.py#L9): [E302](https://duckduckgo.com/?q=pep8%20E302) expected 2 blank lines, found 1
> > [Line 10:12](https://github.com/SwagLyrics/swaglyrics-backend/blob/9ac40293d2ad0b9bebddd95f536a832e8de6f868/test_auth.py#L10): [F405](https://duckduckgo.com/?q=pep8%20F405) 'authenticate_github_app' may be undefined, or defined from star imports: auth
> > [Line 10:49](https://github.com/SwagLyrics/swaglyrics-backend/blob/9ac40293d2ad0b9bebddd95f536a832e8de6f868/test_auth.py#L10): [F405](https://duckduckgo.com/?q=pep8%20F405) 'auth' may be undefined, or defined from star imports: auth
> > [Line 12:1](https://github.com/SwagLyrics/swaglyrics-backend/blob/9ac40293d2ad0b9bebddd95f536a832e8de6f868/test_auth.py#L12): [E305](https://duckduckgo.com/?q=pep8%20E305) expected 2 blank lines after class or function definition, found 1
> > [Line 12:31](https://github.com/SwagLyrics/swaglyrics-backend/blob/9ac40293d2ad0b9bebddd95f536a832e8de6f868/test_auth.py#L12): [W292](https://duckduckgo.com/?q=pep8%20W292) no newline at end of file
> 
> ##### Comment last updated at 2020-01-21 18:30:25 UTC

Please take care of these, also, please add the auth functions to the utils file

**Comment by dhrumilp15** [comment]
Hey! I'm still working on the tests, but as future features are implemented, wouldn't the utils file become too large and unwieldy? Are there future plans to better split the codebase?

**Comment by aadibajpai** [comment]
> Hey! I'm still working on the tests, but as future features are implemented, wouldn't the utils file become too large and unwieldy? Are there future plans to better split the codebase?

Well the utils file houses all utility functions so it is appropriate here. Yeah we might branch off files as the need arises but it should be ok for now.

**Comment by dhrumilp15** [comment]
There were pep8 issues due to the walrus operator in python 3.8, so I rewrote the if statement to not cause pep8 issues in utils.py.

**Comment by aadibajpai** [comment]
> There were pep8 issues due to the walrus operator in python 3.8, so I rewrote the if statement to not cause pep8 issues in utils.py.

Don't edit the code you're not modifying please :)

I _think_ pep8 is configured to only review the diffs so it will be easier for me to review your PR if it only contains changes relevant to the purpose of your PR 🙂

File: issue_maker.py [hunk 1]
```diff
@@ -160,15 +168,29 @@ def create_issue(song, artist, version, stripper='not supported yet'):
...
-        "labels": ["unsupported song"]
-    }
-    r = requests.post('https://api.github.com/repos/SwagLyrics/swaglyrics-for-spotify/issues',
-                      auth=HTTPBasicAuth(username, gh_token), json=json)
-
+        f"stripper -> {stripper}</b>\n\nversion -> {version}</tt>",
+        "labels": ["unsupported song"]}
+
+    token = authenticate_installation()
+    bearer = "token %s" % token
```
Comment (aadibajpai) [inline issue_maker.py]: why not use fstrings :)

File: issue_maker.py [hunk 2]
```diff
@@ -43,10 +43,10 @@
 
 SQLALCHEMY_DATABASE_URI = "mysql+mysqlconnector://{username}:{password}@{username}.mysql.pythonanywhere-services." \
                           "com/{username}${databasename}".format(
-                                                                username=username,
-                                                                password=os.environ['DB_PWD'],
-                                                                databasename="strippers",
-                                                            )
+                              username=username,
```
Comment (aadibajpai) [inline issue_maker.py:53]: Please do not modify code you're not adding or removing :)

File: issue_maker.py [hunk 3]
```diff
@@ -105,7 +110,9 @@ def genius_stripper(song, artist):
     title = f'{song} by {artist}'
     print(f'getting stripper from Genius for {title}')
     url = 'https://api.genius.com/search'
-    headers = {"Authorization": "Bearer {token}".format(token=os.environ['GENIUS'])}
+    headers = {
```
Comment (aadibajpai) [inline issue_maker.py:122]: Please do not modify code you're not adding or removing :)

File: issue_maker.py [hunk 4]
```diff
@@ -187,11 +208,16 @@ def check_song(song, artist):
     """
     global token, t_expiry
     print('using token', token[:41])
-    if t_expiry + 3600 - 300 < time.time():  # check if token expired ( - 300 to add buffer of 5 minutes)
+    if t_expiry + 3600 - \
```
Comment (aadibajpai) [inline issue_maker.py:209]: Please do not modify code you're not adding or removing :)

File: issue_maker.py [hunk 5]
```diff
@@ -187,11 +208,16 @@ def check_song(song, artist):
...
+            300 < time.time():  # check if token expired ( - 300 to add buffer of 5 minutes)
         update_token()
     headers = {"Authorization": f"Bearer {token}"}
-    r = requests.get('https://api.spotify.com/v1/search', headers=headers, params={'q': f'{song} {artist}',
-                                                                                   'type': 'track'})
+    r = requests.get(
```
Comment (aadibajpai) [inline issue_maker.py:213]: Please do not modify code you're not adding or removing :)

File: issue_maker.py [hunk 6]
```diff
@@ -264,7 +290,8 @@ def update():
             issue = create_issue(song, artist, version, stripped)
 
             if issue['status_code'] == 201:
-                print(f'Created issue on the GitHub repo for {song} by {artist}.')
+                print(
```
Comment (aadibajpai) [inline issue_maker.py:292]: Please do not modify code you're not adding or removing :)

File: issue_maker.py [hunk 7]
```diff
@@ -279,7 +306,9 @@ def update():
 def get_stripper():
     song = request.form['song']
     artist = request.form['artist']
-    lyrics = Lyrics.query.filter(Lyrics.song == song).filter(Lyrics.artist == artist).first()
+    lyrics = Lyrics.query.filter(
```
Comment (aadibajpai) [inline issue_maker.py:308]: Please do not modify code you're not adding or removing :)

File: issue_maker.py [hunk 8]
```diff
@@ -411,14 +440,17 @@ def update_webhook():
         pull_info = origin.pull()
 
         if len(pull_info) == 0:
-            return json.dumps({'msg': "Didn't pull any information from remote!"})
+            return json.dumps(
```
Comment (aadibajpai) [inline issue_maker.py:435]: Please do not modify code you're not adding or removing :)

File: issue_maker.py [hunk 9]
```diff
@@ -411,14 +440,17 @@ def update_webhook():
...
+                {'msg': "Didn't pull any information from remote!"})
         if pull_info[0].flags > 128:
-            return json.dumps({'msg': "Didn't pull any information from remote!"})
+            return json.dumps(
```
Comment (aadibajpai) [inline issue_maker.py:438]: Please do not modify code you're not adding or removing :)

File: issue_maker.py [hunk 10]
```diff
@@ -411,14 +440,17 @@ def update_webhook():
...
+                {'msg': "Didn't pull any information from remote!"})
 
         commit_hash = pull_info[0].commit.hexsha
         build_commit = f'build_commit = "{commit_hash}"'
         print(f'{build_commit}')
-        return 'Updated PythonAnywhere server to commit {commit}'.format(commit=commit_hash)
+        return 'Updated PythonAnywhere server to commit {commit}'.format(
```
Comment (aadibajpai) [inline issue_maker.py]: Please do not modify code you're not adding or removing :)

File: utils.py [hunk 11]
```diff
@@ -43,9 +50,11 @@ def decorated_function(*args, **kwargs):
                 if not ua.startswith('GitHub-Hookshot/'):
                     abort(abort_code)
 
-                if not (ip_header := request.headers.get('CF-Connecting-IP')):
+                if not (request.headers.get('CF-Connecting-IP')):
```
Comment (aadibajpai) [inline utils.py]: Please do not modify code you're not adding or removing :)

File: utils.py [hunk 12]
```diff
@@ -55,10 +64,166 @@ def decorated_function(*args, **kwargs):
                     if ip_address(request_ip) in ip_network(block):
                         break
                 else:
-                    print("Unauthorized attempt to deploy by IP {ip}".format(ip=request_ip))
+                    print(
```
Comment (aadibajpai) [inline utils.py]: Please do not modify code you're not adding or removing :)

**Review by aadibajpai — CHANGES_REQUESTED** [review]
Please do not modify code you're not adding or removing :)

Pep8 should apply to the code you're adding, we follow max character limit of 120 chars btw.

File: issue_maker.py [hunk 13]
```diff
@@ -81,8 +81,13 @@ def update_token():
     :return:
     """
     global token, t_expiry
-    r = requests.post('https://accounts.spotify.com/api/token', data={
-        'grant_type': 'client_credentials'}, auth=HTTPBasicAuth(os.environ['C_ID'], os.environ['SECRET']))
+    r = requests.post(
```
Comment (aadibajpai) [inline issue_maker.py:93]: Please do not modify code you're not adding or removing :)

File: utils.py [hunk 14]
```diff
@@ -55,10 +64,166 @@ def decorated_function(*args, **kwargs):
...
+        APP_ID is the id of the github app
+        repo_url is the api url of the repo this app is installed on
+'''
+APP_ID = 51777
+
+SUCCESS = "Epic!"
+
+FAILED = "Oh no!"
+
+repo_url = "https://api.github.com/repos/dhrumilp15/swaglyrics-backend/"
```
Comment (aadibajpai) [inline utils.py]: You probably need to change this

**Comment by dhrumilp15** [comment]
I've written tests and moved the auth code over to utils.py!

**Comment by aadibajpai** [comment]
> I've written tests and moved the auth code over to utils.py!

I don't think you've taken care of my last review 🙂

**Pull request closed without merging**